File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Map<Character, Character> bracketMap = Map.of(
    ')', '(', 
    '}', '{', 
    ']', '['
);
Set<Character> openBrackets = Set.of('(', '{', '[');
Stack<Character> mystack= new Stack<>();
for (char ch : s.toCharArray()) {
if(openBrackets.contains(ch))
{
    mystack.push(ch);
}
else if(bracketMap.containsKey(ch)){
    if(mystack.isEmpty()||mystack.pop()!=bracketMap.get(ch)){
return false;
    }

}

}
return mystack.isEmpty();
}}

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        bracket={"}":"{",")":"(","]":"["}
        res=[]
        for i in s:
            if i in bracket:
                if not res or bracket[i]!=res[-1]:
                    return False
                res.pop()
            else:
                res.append(i)
        return not res

        